FRENCH/src/ch17-02-trait-objects.md

```diff
@@ -0,0 +1,1021 @@
+
+<!--
+We’ll get this error because `String` doesn’t implement the `Draw` trait:
+-->
+
+Nous aurons cette erreur parce que `String` n'implémente pas le trait
+`Affichable` :
+
+<!--
+```text
+error[E0277]: the trait bound `std::string::String: gui::Draw` is not satisfied
+  -- > src/main.rs:7:13
+   |
+ 7 |             Box::new(String::from("Hi")),
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait gui::Draw is not
+   implemented for `std::string::String`
+   |
+   = note: required for the cast to the object type `gui::Draw`
+```
+-->
+
+```text
+error[E0277]: the trait bound `std::string::String: gui::Affichable` is not satisfied
+  -- > src/main.rs:7:13
+   |
+ 7 |             Box::new(String::from("Salut")),
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait gui::Affichable is not
+   implemented for `std::string::String`
+   |
+   = note: required for the cast to the object type `gui::Affichable`
+```
+
+<!--
+This error lets us know that either we’re passing something to `Screen` we
+didn’t mean to pass and we should pass a different type or we should implement
+`Draw` on `String` so that `Screen` is able to call `draw` on it.
+-->
+
+L'erreur nous fait savoir que soit nous passons quelque chose à `Ecran` que nous
+ne voulions pas lui passer et nous devrions lui passer un type différent, soit
+nous devrions implémenter `Affichable` sur `String` de sorte que `Ecran` puisse
+appeler `afficher` dessus.
+
+<!--
+### Trait Objects Perform Dynamic Dispatch
+-->
+
+### Les objets traits effectuent du dispatch dynamique
+
+<!--
+Recall in the [“Performance of Code Using
+Generics”][performance-of-code-using-generics]<!-- ignore -- > section in
+Chapter 10 our discussion on the monomorphization process performed by the
+compiler when we use trait bounds on generics: the compiler generates
+nongeneric implementations of functions and methods for each concrete type
+that we use in place of a generic type parameter. The code that results from
+monomorphization is doing *static dispatch*, which is when the compiler knows
+what method you’re calling at compile time. This is opposed to *dynamic
+dispatch*, which is when the compiler can’t tell at compile time which method
+you’re calling. In dynamic dispatch cases, the compiler emits code that at
+runtime will figure out which method to call.
+-->
+
+Rappelez-vous de notre discussion dans la section [“Performance du code
+utilisant les génériques”][performance-of-code-using-generics]<!-- ignore --> du
+chapitre 10 à propos du processus de monomorphisation effectué par le
+compilateur quand nous utilisons des traits liés sur des génériques : le
+compilateur génère des implémentations non génériques de fonctions et de
+méthodes pour chaque type concret que nous utilisons à la place d'un paramètre
+de type générique. Le code résultant de la monomorphisation effectue du
+*dispatch statique*, c'est-à-dire quand le compilateur sait quelle méthode
```

Comment:
OK for translating *dispatch* as `répartition`, but I think it will would be best to specify the English term in parentheses when the term is first encountered in the text:

> de la *répartition statique* (*static dispatch*), c'est-à-dire…